src/tools/abfe.py

```diff
@@ -1,621 +1,905 @@
 """this module contains various functions to run steps of an ABFE workflow"""
 
-from typing import Literal
-
+import ast
+import importlib.resources
+import json
+import os
+from dataclasses import dataclass, field, fields
+from pathlib import Path
+from typing import Any, Callable, List, Optional, Union
+
+import numpy as np
+import pandas as pd
 from beartype import beartype
+from box import Box
+from deeporigin import chemistry
 from deeporigin.data_hub import api
-from deeporigin.exceptions import DeepOriginException
 from deeporigin.tools import run
 from deeporigin.tools.toolkit import _ensure_columns, _ensure_database
+from deeporigin.tools.utils import query_run_status
 from deeporigin.utils.config import construct_resource_url
+from deeporigin.utils.notebook import render_mermaid
+from IPython.display import HTML, display
 
 # constants and types
 ABFE_DB = "ABFE"
+COL_DELTA_G = "FEP ΔG (kcal/mol)"
+COL_LIGAND_FILE = "ligand_file"
+COL_PROTEIN_FILE = "protein_file"
+COL_COMPLEX_PREP_OUTPUT = "complex_prep_output"
+COL_LIGAND_PREP_OUTPUT = "ligand_prep_output"
+COL_EMEQ_OUTPUT = "emeq_output"
+COL_SOLVATION_FEP_OUTPUT = "solvation_fep_output"
+COL_MD_OUTPUT = "md_output"
+COL_BINDING_FEP_OUTPUT = "binding_fep_output"
+COL_END_TO_END_OUTPUT = "end_to_end_output"
+COL_JOB_IDS = "job_ids"
+
+
+class PrettyDict(Box):
+    """A dict subclass with a custom pretty-print representation."""
+
+    def __repr__(self):
+        """pretty print a dict"""
+        return json.dumps(
+            dict(self),
+            indent=2,
+            ensure_ascii=False,
+        )
+
+    def _repr_html_(self):
+        """pretty print a dict"""
+        self.__repr__()
 
-charge_methods = Literal["gas", "bcc"]
-"""Available charge methods"""
 
-ligand_force_fields = Literal["gaff", "gaff2", "openff"]
-"""Available ligand force fields. `gaff` is General Amber Force Field, gaff2 is an updated version of gaff. """
+@beartype
+def _load_params(step: str) -> Box:
+    """load default values for abfe end to end run"""
 
-force_fields = Literal["ff14SB", "ff99SB-ildn"]
-"""Available force fields"""
+    with importlib.resources.open_text("deeporigin.json", f"{step}.json") as f:
+        return PrettyDict(json.load(f))
 
-integrators = Literal[
-    "BAOABIntegrator",
-    "LangevinIntegrator",
-    "SteepestDescentMinimizer",
-    "LeapFrogIntegrator",
-    "SoluteSolventSplittingIntegrator",
-]
-"""Integrator available for simulation"""
 
+@beartype
+def _ensure_db_for_abfe() -> dict:
+    """ensure that there is a database for FEP on Data Hub"""
 
-prod_md_defaults = {
-    "integrator": "BAOABIntegrator",
-    "Δt": 0.004,
-    "T": 298.15,
-    "cutoff": 0.9,
-    "fourier_spacing": 0.12,
-    "hydrogen_mass": 2,
-    "barostat": "MonteCarloBarostat",
-    "barostat_exchange_interval": 500,
-}
```

Comment:
this is all gone because it exists as a dict that's read from a params.json file